test/test_nn.py

```diff
@@ -2954,6 +2954,35 @@ def test_threshold_int(self):
         expected = torch.tensor([99, 99, 99, 99, 1, 2, 3])
         self.assertEqual(F.threshold(x, 0, 99), expected)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_embedding_max_norm_unsorted_repeating_indices(self):
+        def create_embedding(device):
+            # Seed RNG so we get the same Embedding each time
+            torch.manual_seed(0)
+            return torch.nn.Embedding(
+                num_embeddings=20,
+                embedding_dim=64,
+                max_norm=1.0).to(device)
+
+        ix = torch.arange(2, device='cpu', dtype=torch.long).repeat(2000)
+        out_cpu = create_embedding('cpu')(ix)
+
+        ix = ix.to('cuda')
+        out = create_embedding('cuda')(ix)
+        self.assertEqual(out.cpu(), out_cpu)
+
+        # Since CUDA Embedding internally sorts indices, we should
+        # be able to get the same result by sorting the indices before
+        # giving them to Embedding. Then we can reorder the output
+        # to match the order of the unsorted indices
+        ix_sorted, ix_orig_indices = ix.sort()
+        ix_orig_indices = ix_orig_indices.tolist()
+        ix_reverse_sort_indices = [None] * len(ix_orig_indices)
+        for ind, orig_ind in enumerate(ix_orig_indices):
+            ix_reverse_sort_indices[orig_ind] = ind
+        out_presorted = create_embedding('cuda')(ix_sorted)[torch.tensor(ix_reverse_sort_indices, device='cuda')]
+        self.assertEqual(out, out_presorted)
```

Comment:
I'm not too sure if this test case is very important. I can remove it if not.

Reply:
Yeah, please remove it, just comparing results of inputs with repeating indices is enough.

Reply:
Done